File: 0001. Two Sum.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
    
    for i, num := range nums {
        num2 := target - num
        i2, ok := m[num2]
        
        if ok {
            return []int{i, i2}
        } else {
            m[num] = i
        }
    }
    
    return []int{-1, -1}
}

File: yelp/0001. Two Sum.py
from typing import *

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        val_idx = {}
        
        for i in range(n):
            complement = target - nums[i]
            
            if complement in val_idx:
                return [val_idx[complement], i]
            
            val_idx[nums[i]] = i
        
        return [-1, -1]
